File: src/code/python/sorting_algorithms/sleep_sort.py
from threading import Thread
from time import sleep


def sleep_sort(arr: list ) -> list:
    sorted_arr = []
    threads = []

    for num in arr:
        thread = Thread(target=snorlax, args=(num, sorted_arr))
        threads.append(thread)

    for thread in threads:
        thread.start( )

    for thread in threads:
        thread.join()

    return sorted_arr

def snorlax(num: float, arr: list) -> None:
    sleep(num / 1000.0)
    arr.append(num)

File: src/code/ruby/sorting_algorithms/sleep_sort.rb
def sleep_sort(arr)
  sorted_arr  = []
  threads     = []

  arr.each do |num|
    threads << Thread.new do
      sleep(num / 1000.0)
      sorted_arr << num
    end
  end

  threads.each(&:join)
  sorted_arr
end
